### PR TITLE
fix sorting for iterables that define copymutable

### PR DESCRIPTION
**Pull request by vtjnash**
`copymutable` is only defined to return an array for abstract arrays, but that is only what this method is never called with. For other types, it has a default of `collect` (which means the default behavior appears to give an array), but can be changed by other types (such as AbstractSet) to do something different.

Correction to #46104

### CODE REVIEW

**Comment by LilithHafner** [comment]
Since we only support sorting abstract arrays with `sort!` (and that's all we _can_ support because we need length and random access getindex and setindex) this seems reasonable.

This would be inconvenient for iterable and sortable non-abstractarray types that bring their own `copymutable` and `sort!` methods. However, in that case all they have to do is define `Base.sort(x::MyType; kw...) = sort!(copymutable(x); kw...)`, so not a big loss.

`collect` isn't perrfect; we want `collect_to_abstractarray`. However, if we ever do add `collect_to_abstractarray`, I think we could switch this as a minor or non-breaking change.

**Review by LilithHafner — APPROVED** [review]
LGTM, though we should probably wait to see if triage decides to revert #46104 before we merge this.

**Comment by vtjnash** [comment]
We don't need triage to approve merging a bugfix, even if there may be later changes to the API

**Pull request merged**
